packages/wallets/src/components/DatePicker/DatePicker.tsx

```diff
@@ -42,61 +40,64 @@ const DatePicker = ({
         setIsCalendarOpen(false);
     });
 
-    const handleDateChange: CalendarProps['onChange'] = value => {
-        const calendarSelectedDate = Array.isArray(value) ? value[0] : value;
-        setSelectedDate(calendarSelectedDate);
-        setIsCalendarOpen(false);
-    };
-
-    useEffect(() => {
-        if (selectedDate !== null) {
-            onDateChange(unixToDateString(selectedDate));
-        }
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [selectedDate]);
-
     return (
-        <div className='wallets-datepicker' ref={datePickerRef}>
-            <FlowTextField
-                disabled={disabled}
-                inputMode='none'
-                label={label}
-                message={message}
-                name={name}
-                onClick={toggleCalendar}
-                onKeyDown={e => e.preventDefault()}
-                ref={inputeDateRef}
-                renderRightIcon={() => (
-                    <button
-                        className='wallets-datepicker__button'
-                        data-testid='wallets_datepicker_button'
-                        disabled={disabled}
-                        onClick={toggleCalendar}
-                    >
-                        <LegacyCalendar1pxIcon iconSize='xs' />
-                    </button>
-                )}
-                showMessage
-                type='text'
-                validationSchema={validationSchema}
-                value={selectedDate !== null ? unixToDateString(selectedDate, displayFormat) : ''}
-            />
-            {isCalendarOpen && (
-                <div
-                    className={`wallets-datepicker__container wallets-datepicker__container--${mobileAlignment}`}
-                    data-testid='wallets_datepicker_container'
-                >
-                    <Calendar
-                        formatShortWeekday={customFormatShortWeekday}
-                        inputRef={inputeDateRef}
-                        maxDate={maxDate}
-                        minDate={minDate}
-                        onChange={handleDateChange}
-                        value={selectedDate !== null ? selectedDate : ''}
-                    />
-                </div>
-            )}
-        </div>
+        <Field name={name}>
+            {({ field, form }: FieldProps) => {
+                return (
+                    <div className='wallets-datepicker' ref={datePickerRef}>
+                        <WalletTextField
```

Comment:
Is there a reason why the DatePicker is using a WalletTextField component wrapped in a Formik field instead of extending the FormField component?

Reply:
Since DatePicker is a much more generic component which can used in other places in future, I didn't want to add FormField to it. There is a possibility that we can move it to our UI library as well.